### PR TITLE
fix(authenticator.utils): get ID_KEY from authenticator configuration, not defaults

### PR DESCRIPTION
**Pull request by BrennanPaciorek**
- **tests(authentication): fix UID_KEY tests for authentication.determine_username_from_uid_social, introduce failing test**
- **fix(authenticator.utils): get ID_KEY from authenticator configuration, not defaults**

AAP-37203

### CODE REVIEW

File: ansible_base/authentication/utils/authentication.py [hunk 1]
```diff
@@ -95,13 +95,14 @@ def check_system_username(uid: str) -> None:
 
 
 def determine_username_from_uid_social(**kwargs) -> dict:
-    uid_field = getattr(kwargs.get('backend', None), 'ID_KEY', 'username')
-    if uid_field is None:
-        uid_field = 'username'
+    backend = kwargs.get('backend', None)
+    uid_field = 'username'
+    if backend and hasattr(backend, "setting"):
+        uid_field = backend.setting('ID_KEY', default='username')
```
Comment (john-westcott-iv) [inline ansible_base/authentication/utils/authentication.py]: Is there a case where `ID_KEY` might be directly in backend that we need to cover as well?

Comment (john-westcott-iv) [reply ansible_base/authentication/utils/authentication.py]: For example: https://github.com/python-social-auth/social-core/blob/master/social_core/backends/battlenet.py

Comment (BrennanPaciorek) [reply ansible_base/authentication/utils/authentication.py]: You are correct. Updating this now with more favorable logic. This one setting fetch seems to have a ton of cases to consider.

File: ansible_base/authentication/utils/authentication.py [hunk 2]
```diff
@@ -95,13 +95,21 @@ def check_system_username(uid: str) -> None:
...
 
 def determine_username_from_uid_social(**kwargs) -> dict:
-    uid_field = getattr(kwargs.get('backend', None), 'ID_KEY', 'username')
-    if uid_field is None:
-        uid_field = 'username'
+    backend = kwargs.get('backend', None)
+    uid_field = 'username'
+    if backend:
+        # Update our fallback if the authenticator has an ID_KEY field
+        if hasattr(backend, "ID_KEY"):
```
Comment (BrennanPaciorek) [inline ansible_base/authentication/utils/authentication.py]:
```suggestion
        # Update our fallback if the authenticator has an ID_KEY field, ignore if ID_KEY is None
        if getattr(backend, "ID_KEY", None) is not None:
```
This is probably something we want to do.

**Review by john-westcott-iv — APPROVED** [review]

**Pull request merged**